src/libs/ascent/runtimes/flow_filters/ascent_runtime_vtkh_filters.cpp

```diff
@@ -5246,6 +5247,133 @@ VTKHVTKFileExtract::execute()
 }
 
 
+//-----------------------------------------------------------------------------
+
+VTKHMIR::VTKHMIR()
+:Filter()
+{
+// empty
+}
+
+//-----------------------------------------------------------------------------
+VTKHMIR::~VTKHMIR()
+{
+// empty
+}
+
+//-----------------------------------------------------------------------------
+void
+VTKHMIR::declare_interface(Node &i)
+{
+    i["type_name"]   = "vtkh_mir";
+    i["port_names"].append() = "in";
+    i["output_port"] = "true";
+}
+
+//-----------------------------------------------------------------------------
+bool
+VTKHMIR::verify_params(const conduit::Node &params,
+                        conduit::Node &info)
+{
+    info.reset();
+
+    bool res = check_string("matset",params, info, true);
+    res &= check_numeric("error_scaling", params, info, false);
+    res &= check_numeric("scaling_decay", params, info, false);
+    res &= check_numeric("iterations", params, info, false);
+    res &= check_numeric("max_error", params, info, false);
+
+    std::vector<std::string> valid_paths;
+    valid_paths.push_back("matset");
+    valid_paths.push_back("error_scaling");
+    valid_paths.push_back("scaling_decay");
+    valid_paths.push_back("iterations");
+    valid_paths.push_back("max_error");
+
+    std::string surprises = surprise_check(valid_paths, params);
+
+    if(surprises != "")
+    {
+      res = false;
+      info["errors"].append() = surprises;
+    }
+
+    return res;
+}
+
+//-----------------------------------------------------------------------------
+void
+VTKHMIR::execute()
+{
+
+    if(!input(0).check_type<DataObject>())
+    {
+        ASCENT_ERROR("vtkh_MIR input must be a data object");
+    }
+
+    DataObject *data_object = input<DataObject>(0);
+    if(!data_object->is_valid())
+    {
+      set_output<DataObject>(data_object);
+      return;
+    }
+    std::shared_ptr<VTKHCollection> collection = data_object->as_vtkh_collection();
+
+    std::string matset_name = params()["matset"].as_string();
+    std::string ids_name = "material_ids";//matset_name + "_ids";
+    if(!collection->has_field(ids_name))
+    {
+      bool throw_error = false;
+      detail::field_error(ids_name, this->name(), collection, throw_error);
+      // this creates a data object with an invalid soource
+      set_output<DataObject>(new DataObject());
+      return;
+    }
+
+    std::string topo_name = collection->field_topology(ids_name);
+
+    vtkh::DataSet &data = collection->dataset_by_topology(topo_name);
+    if(!graph().workspace().registry().has_entry("is_discrete"))
+    {
+      
+      int *is_discrete = new int();
+      *is_discrete = 1;
+      graph().workspace().registry().add<int>("is_discrete",
+                                               is_discrete,
```

Comment:
wondering if there is another way to do this hand shake?

Reply:
Stripped this out and now it's a param for pseudocolor plots